### ISSUE TITLE
Wrong escaping function, using esc_attr_e()

### ISSUE THREAD

**Issue by farhadsakhaei** [Automattic/VIP-Coding-Standards#807]
My code:

`<label for="<?php echo esc_attr( $this->get_field_id( 'display-mode' ) ); ?>">`

or

`<label for="<?php esc_attr_e( $this->get_field_id( 'display-mode' ) ); ?>">`

**I get this error:**

`Wrong escaping function, using `esc_attr_e()` in a context outside of HTML attributes may not escape properly.PHPCS(WordPressVIPMinimum.Security.ProperEscapingFunction.notAttrEscAttr)`

**Comment by farhadsakhaei** [comment]
@GaryJones 
Hi, Thank you for reply,
this is my full line of code:

`<label for="<?php echo esc_attr( $this->get_field_id( 'display-mode' ) ); ?>"><?php esc_attr_e( 'Display Mode:', 'aparat-feed' ); ?></label>`

![4PoiabZsCU](https://github.com/Automattic/VIP-Coding-Standards/assets/33499600/84794a90-fdcc-45eb-a895-c733d791f5bf)

**Comment by jrfnl** [comment]
@farhadsakhaei Your error is for the `<?php esc_attr_e( 'Display Mode:', 'aparat-feed' ); ?>` and the sniff is correct  as that's HTML context, not attribute context.

**Comment by farhadsakhaei** [comment]
@jrfnl 

Yes, You are right, That was because the pointer was pointing another tag
Thank you

**Comment by jrfnl** [comment]
@farhadsakhaei When in doubt, you can use the `--report=code` option to see which bit of the code is being flagged by which sniff. Hope that helps.

Shall we close this issue ?

**Comment by farhadsakhaei** [comment]
@jrfnl 
Thank you so much for help and recommendation
Yes, of course
Sincerly
